### PR TITLE
Ways to speed up NPM installs

### PR DESCRIPTION
**Pull request by hx-markterry**
#### What

Adding some best practices around creation of node modules.
#### Why

Our build time and size for our tripapp projects is way too big, adhering to these points should help keep this under control.
#### Reviewers

**Review 1**
- [x] :+1:

**Review 2** *
- [x] :+1:

### CODE REVIEW

File: general-javascript-best-practices.md [hunk 1]
```diff
@@ -209,3 +209,24 @@ var name = "oli";
...
+
+## NPM Modules
+
+When creating an NPM module for others to use (private or open source) it is worth giving consideration to the following points.
+
+### Post install scripts
+
+These are run every time the module is installed by a consuming project which will increase CI and developer build time.
+
+If the module is being released to NPM consider using a `prerelease` script instead.
```
Comment (shackpank) [inline general-javascript-best-practices.md]: `prepublish` ?

File: general-javascript-best-practices.md [hunk 2]
```diff
@@ -209,3 +209,24 @@ var name = "oli";
...
+
+For internal modules a separate `build` task can be used then the module packed and uploaded as an GitHub asset to accompany releases.
+
+### Dependencies
+
+When specifying requirements first aim to use the `^` semver match. Only resort to using `~` or pinning the version if there is a known problem the module's
```
Comment (shackpank) [inline general-javascript-best-practices.md]: should be "a known problem with", and unnecessary newline?

File: general-javascript-best-practices.md [hunk 3]
```diff
@@ -209,3 +209,24 @@ var name = "oli";
...
+
+When creating an NPM module for others to use (private or open source) it is worth giving consideration to the following points.
+
+### Post install scripts
+
+These are run every time the module is installed by a consuming project which will increase CI and developer build time.
+
+If the module is being released to NPM consider using a `prerelease` script instead.
+
+For internal modules a separate `build` task can be used then the module packed and uploaded as an GitHub asset to accompany releases.
```
Comment (shackpank) [inline general-javascript-best-practices.md]: Should this point you to the deployment helper we're already using (as it's open-sourced) for a more detailed howto?

**Comment by shackpank** [comment]
This could be cross-linked with [publishing to npm](https://github.com/holidayextras/culture/blob/master/publishing-to-npm.md)

**Comment by shackpank** [comment]
Couple of comments but overall :+1:

**Comment by hx-markterry** [comment]
@shackpank have now addressed your comments, cheers.

**Comment by lightningtommo** [comment]
Any step towards eliminating duplication is a good thing, for developer and customer.

**Pull request merged**